src/Polly.SharedSpecs/Timeout/TimeoutSpecs.cs

```diff
@@ -139,6 +139,24 @@ public void Should_throw_when_timeout_is_less_than_execution_duration__pessimist
         }
 
         [Fact]
+        public void Should_rethrow_exception_from_inside_delegate__pessimistic()
+        {
+            var policy = Policy.Timeout(TimeSpan.FromMilliseconds(50), TimeoutStrategy.Pessimistic);
+
+            policy.Invoking(p => p.Execute(() => { throw new NotImplementedException(); })).ShouldThrow<NotImplementedException>();
+        }
+
+        [Fact]
+        public void Should_rethrow_aggregate_exception_from_inside_delegate__pessimistic()
+        {
+            var policy = Policy.Timeout(TimeSpan.FromMilliseconds(50), TimeoutStrategy.Pessimistic);
+            var msg = "Aggregate Exception thrown from the delegate";
+            // Check to see if nested aggregate exceptions are unwrapped correctly
+            AggregateException exception = new AggregateException(msg, new NotImplementedException());
+            policy.Invoking(p => p.Execute(() => { throw exception; })).ShouldThrow<AggregateException>().WithMessage(msg).WithInnerException<NotImplementedException>();
+        }
+
```

Comment:
@hambudi Your spec here provides perfect coverage to demonstrate that an aggregate exception thrown within the Policy is correctly unwrapped.  

I built on yours to add couple more specs which demonstrate the cases when the `AggregateException` thrown within the user delegate contains multiple `InnerExceptions`.  Would you be happy to add these (if you agree they are useful) to your PR?

```c#

        [Fact]
        public void Should_rethrow_aggregate_exception_with_multiple_exceptions_from_inside_delegate__pessimistic()
        {
            var policy = Policy.Timeout(TimeSpan.FromMilliseconds(50), TimeoutStrategy.Pessimistic);
            var msg = "Aggregate Exception thrown from the delegate";

            Exception innerException1 = new NotImplementedException();
            Exception innerException2 = new DivideByZeroException();
            AggregateException aggregateException = new AggregateException(msg, innerException1, innerException2);
            Action action = () => { throw aggregateException; };

            // Whether executing the delegate directly, or through the policy, exception behaviour should be the same.
            action.ShouldThrow<AggregateException>()
                .WithMessage(msg)
                .And.InnerExceptions.Should().BeEquivalentTo<Exception>(new[] {innerException1, innerException2});

            policy.Invoking(p => p.Execute(action)).ShouldThrow<AggregateException>()
                .WithMessage(msg)
                .And.InnerExceptions.Should().BeEquivalentTo<Exception>(new[] {innerException1, innerException2});
        }

        [Fact]
        public void Should_rethrow_aggregate_exception_with_example_cause_of_multiple_exceptions_from_inside_delegate__pessimistic()
        {
            var policy = Policy.Timeout(TimeSpan.FromMilliseconds(50), TimeoutStrategy.Pessimistic);

            Exception innerException1 = new NotImplementedException();
            Exception innerException2 = new DivideByZeroException();
            Action action = () =>
            {
                Task task1 = Task.Run(() => { throw innerException1; });
                Task task2 = Task.Run(() => { throw innerException2; });
                Task.WhenAll(task1, task2).Wait();
            };

            // Whether executing the delegate directly, or through the policy, exception behaviour should be the same.
            action.ShouldThrow<AggregateException>()
                .And.InnerExceptions.Should().BeEquivalentTo<Exception>(new[] {innerException1, innerException2});

            policy.Invoking(p => p.Execute(action)).ShouldThrow<AggregateException>()
                .And.InnerExceptions.Should().BeEquivalentTo<Exception>(new[] {innerException1, innerException2});
        }

        [Fact]
        public void Should_rethrow_aggregate_exception_with_another_example_cause_of_multiple_exceptions_from_inside_delegate__pessimistic()
        {
            var policy = Policy.Timeout(TimeSpan.FromMilliseconds(50), TimeoutStrategy.Pessimistic);

            Exception innerException1 = new NotImplementedException();
            Exception innerException2 = new DivideByZeroException();
            Action action = () =>
            {
                Action action1 = () => { throw innerException1; };
                Action action2 = () => { throw innerException2; };
                Parallel.Invoke(action1, action2);
            };

            // Whether executing the delegate directly, or through the policy, exception behaviour should be the same.
            action.ShouldThrow<AggregateException>()
                .And.InnerExceptions.Should().BeEquivalentTo<Exception>(new[] { innerException1, innerException2 });

            policy.Invoking(p => p.Execute(action)).ShouldThrow<AggregateException>()
                .And.InnerExceptions.Should().BeEquivalentTo<Exception>(new[] { innerException1, innerException2 });
        }
```

@udlose These specs demo what I mentioned in previous comment: if the delegate executed through `TimeoutPolicy` in `TimeoutStrategy.Pessimistic` throws an `AggregateException -> (multiple inner exceptions)`, that get's expressed _within_ Polly (due to the way this strategy has to execute the delegate) as `AggregateException -> AggregateException -> (multiple inner exceptions)`, and @hambudi 's PR correctly strips off the outer `AggregateException`.